utilities/pipelines/staticValidation/compliance/module.tests.ps1

```diff
@@ -192,6 +205,145 @@ Describe 'File/folder tests' -Tag 'Modules' {
                 $pathExisting | Should -Be $false -Because ('The module is not orphaned but owned by [{0}].' -f $relevantCSVRow.PrimaryModuleOwnerGHHandle)
             }
         }
+
+        # Changelogs are required for all (child-)modules that are published (having a version.json file)
+        It '[<moduleFolderName>] Module should contain a [` CHANGELOG.md `] file.' -TestCases ($moduleFolderTestCases | Where-Object { $_.moduleVersionExists }) {
+
+            param(
+                [string] $moduleFolderPath,
+                [bool] $moduleVersionExists
+            )
+
+            $changelogFilePath = Join-Path -Path $moduleFolderPath 'CHANGELOG.md'
+            $pathExisting = Test-Path $changelogFilePath
+            $pathExisting | Should -Be $true
+        }
+
+        It '[<moduleFolderName>] `CHANGELOG.md` must contain a section with content and versions are ordered descending.' -TestCases ($moduleFolderTestCases | Where-Object { $_.moduleVersionExists }) {
+
+            param(
+                [string] $moduleFolderPath,
+                [bool] $moduleVersionExists
+            )
+
+            $changelogFilePath = Join-Path -Path $moduleFolderPath 'CHANGELOG.md'
+            if (-not (Test-Path -Path $changelogFilePath)) {
+                Set-ItResult -Skipped -Because 'CHANGELOG.md file not found.'
+                return
+            }
+
+            # get the version from the version.json file. If the main.json version did not change, use the last published version to be able to add missing entries to the changelog
+            $newModuleVersion = Get-ModuleTargetVersion -ModuleFolderPath $moduleFolderPath -CompareJsonVersion $true
+
+            $changelogContent = Get-Content $changelogFilePath
+            $sections = $changelogContent | Where-Object { $_ -match '^##\s+' }
+            $changelogSection = $sections | Where-Object { $_ -match "^##\s+$newModuleVersion" }
+
+            # the changelog should not be empty
```

Comment:
I'd suggest having a separate `It` block for each of the below paragraphs (#). We can think of having a dedicated `Context` or even `Describe` block if the case, similarly to what we do for readmes

Reply:
Resolving as done